File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) string {
	var rna []rune

    for _, nucleotide := range dna {
        switch nucleotide {
            case 'G':
            	rna = append(rna, 'C')
            case 'C':
            	rna = append(rna, 'G')
            case 'T':
            	rna = append(rna, 'A')
            case 'A':
            	rna = append(rna, 'U')
        }
    }

    return string(rna)
}

File: solutions/typescript/rna-transcription/1/rna-transcription.ts
export function toRna(dna: string): string {
  const mapping: Record<string, string> = {
    G: 'C',
    C: 'G',
    T: 'A',
    A: 'U'
  };

  return dna
    .split('')
    .map((nucleotide: string): string => {
      if (!(nucleotide in mapping)) {
        throw new Error('Invalid input DNA.');
      }
      return mapping[nucleotide];
    })
    .join('');
}
